tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/SendMessageAsyncIntegration.cs

```diff
@@ -52,16 +52,24 @@ public class SendMessageAsyncIntegration
             // for the InjectHeadersIntoMessage<TSendMessageRequest> call below
             var requestProxy = request.DuckCast<ISendMessageRequest>();
 
+            var queueName = AwsSqsCommon.GetQueueName(requestProxy.QueueUrl);
             var scope = AwsSqsCommon.CreateScope(Tracer.Instance, Operation, out var tags, spanKind: SpanKinds.Producer);
             if (tags is not null && requestProxy.QueueUrl is not null)
             {
                 tags.QueueUrl = requestProxy.QueueUrl;
-                tags.QueueName = AwsSqsCommon.GetQueueName(requestProxy.QueueUrl);
+                tags.QueueName = queueName;
             }
 
             if (scope?.Span.Context != null)
             {
-                ContextPropagation.InjectHeadersIntoMessage<TSendMessageRequest>(requestProxy, scope.Span.Context);
+                var dataStreamsManager = Tracer.Instance.TracerManager.DataStreamsManager;
+                if (dataStreamsManager != null && dataStreamsManager.IsEnabled)
```

Comment:
`queueName` could be null, so maybe you want to skip the DSM entirely in that case?
```suggestion
                if (!string.IsNullOrEmpty(queueName)
                  && Tracer.Instance.TracerManager.DataStreamsManager is { IsEnabled: true } dataStreamsManager)
```